aten/src/ATen/TensorImpl.cpp

```diff
@@ -62,13 +85,17 @@ TensorImpl::TensorImpl(Storage&& storage, TensorTypeId type_id, ScalarType scala
       numel_(0),
       type_id_(type_id),
       scalar_type_(scalar_type),
-      is_variable_(is_variable) {}
+      is_variable_(is_variable),
+      options_(options) {}
 
 IntList TensorImpl::sizes() const {
   return sizes_;
 }
 
 IntList TensorImpl::strides() const {
+  if (!options_.has_strides_) {
+    AT_ERROR("This type of tensor does not have strides");
```

Comment:
This error message could be improved a bit I think -- what's a "type of tensor"?  That's never defined.  I'd say something like "Tensor does not support strides()".

Reply:
Or use TensorTypeId :).